types/currency.go

```diff
@@ -89,54 +111,66 @@ func (c Currency) Sqrt() (y Currency) {
 	return
 }
 
-// RoundDown returns the largest multiple of n <= c.
-func (c Currency) RoundDown(n uint64) (y Currency) {
-	diff := new(big.Int).Mod(&c.i, new(big.Int).SetUint64(n))
-	y.i.Sub(&c.i, diff)
+// Sub returns a new Currency value y = c - x. Behavior is undefined when
+// c < x.
+func (c Currency) Sub(x Currency) (y Currency) {
+	if c.Cmp(x) < 0 {
+		y = c
+		if DEBUG {
+			panic("subtraction resulted in negative currency")
+		}
+	} else {
+		y.i.Sub(&c.i, &x.i)
+	}
 	return
 }
 
-// String implements the fmt.Stringer interface.
-func (c Currency) String() string {
-	return c.i.String()
+// MarshalJSON implements the json.Marshaler interface.
+func (c Currency) MarshalJSON() ([]byte, error) {
+	return c.i.MarshalJSON()
 }
 
-// MarshalSia implements the encoding.SiaMarshaler interface. It returns the
-// byte-slice representation of the Currency's internal big.Int, prepended
-// with a single byte indicating the length of the slice. This implies a
-// maximum encodable value of 2^(255 * 8), or approximately 10^614.
-//
-// Note that as the bytes of the big.Int correspond to the absolute value of
-// the integer, there is no way to marshal a negative Currency.
-func (c Currency) MarshalSia() []byte {
-	b := c.i.Bytes()
-	return append(
-		[]byte{byte(len(b))},
-		b...,
-	)
+// UnmarshalJSON implements the json.Unmarshaler interface. An error is
+// returned if a negative number is provided.
+func (c *Currency) UnmarshalJSON(b []byte) error {
+	err := c.i.UnmarshalJSON(b)
+	if err != nil {
+		return err
+	}
+	if c.Cmp(ZeroCurrency) < 0 {
```

Comment:
There is no Sign() function anymore since it can't ever be negative.

I guess I should add it back though. Cmp(ZeroCurrency) is a pretty awful construction, especially when you have to import the types package just to do that.